src/wp-includes/widgets.php

```diff
@@ -1251,13 +1251,30 @@ function _wp_sidebars_changed() {
 		$sidebars_widgets = wp_get_sidebars_widgets();
 	}
 
-	retrieve_widgets( true );
+	sync_registered_widgets( true );
 }
 
 /**
- * Look for "lost" widgets, this has to run at least on each theme change.
+ * Do not use, deprecated.
+ *
+ * Use sync_registered_widgets() instead.
  *
  * @since 2.8.0
+ * @deprecated 5.8.1 Use sync_registered_widgets()
+ * @see sync_registered_widgets()
+ *
+ * @param string|bool $theme_changed
+ * @return array
+ */
+function retrieve_widgets( $theme_changed = false ) {
+	return sync_registered_widgets( $theme_changed );
```

Comment:
```suggestion
         _deprecated_function( __FUNCTION__, '5.8.1, 'retrieve_widgets' );
	return sync_registered_widgets( $theme_changed );
```

We should add a _deprecated_function function call here.